linode/manager
# refactor: [M3-8720] - One `ImageSelect` to rule them all

## Description 📝
This PR consolidates the three different ImageSelect we had in the codebase.
1. src/components/ImageSelectV2
2. src/components/ImageSelect (legacy)
3. src/features/Images/ImageSelect (used in `StackscriptCreate` for instance)

It unifies the experience for every feature where an image picker is featured and cleans up the codebase accordingly from unnecessary utils.

## Changes  🔄
- Consolidate ImageSelect to include treatment, styling and interfaces of all ImageSelect components, including the "All/Any" and multiple options
- Fix sorting of groups and options
  - Groups are sorted alphabetically, with the exception of private images ("My Images") being first
  - Within groups, options are sorted by the `created` property (latests release first)
- Get rid of console warnings for 
  - `ImageSelect`
  - `Dialog` (unrelated but caught while testing)
- e2e updates to look for the new Autocomplete
- unit coverage updates and improvements
- cleanup

## Target release date 🗓️

| Before (stackscript create, for instance)  | After   |
| ------- | ------- |
| ![Screenshot 2024-10-22 at 14 01 03](https://github.com/user-attachments/assets/8b4207d9-5b97-4133-ae8c-5a55b692cc81) | ![Screenshot 2024-10-22 at 14 00 40](https://github.com/user-attachments/assets/a8d2f1df-23a6-4623-8af1-4254d5cc0462) |

## How to test 🧪

### Prerequisites

### Verification steps
Verify the behavior and content of the following UIs while comparing to their production equivalent:
- `/linode/create` (no changes)
- `/linodes/create?type=Images` (no changes)
- `/linodes` > Action menu > Rebuild
  - rebuild from image (no changes)
  - rebuild from stackscript (no changes)
- `linode/{id}/storage` (linode must be powered down) > Add Disk > Create from Image (old grouping replaced)
- `stackscripts/create` (old grouping replaced)

## As an Author I have considered 🤔

*Check all that apply*
